lib/Labyrinth/Players.hs

```diff
@@ -15,23 +16,40 @@ module Labyrinth.Players
 import           Prelude hiding (lookup)
 import           Lens.Micro     ((^.))
 import           Lens.Micro.TH  (makeLenses)
-import qualified Data.Map as M
+import           Data.Map       (Map)
+import qualified Data.Map       as M
 
 type Name = String
-data Color = Yellow | Blue | Green | Red deriving (Show, Eq, Ord)
+type PlayerStore = Map Color Player
+data Color = Clear | Yellow | Blue | Green | Red deriving (Show, Eq, Ord)
```

Comment:
I didn't dwell into it very much, but GHG started moaning about a missing `Monoid`, so I made it and hid the constructor for `mempty`. Maybe a clear sign I don't really know what I'm doing.